### PR TITLE
Prepare for Connectbox interface

### PR DESCRIPTION
**Pull request by edwinsteele**
@furnox is going to integrate the client side connect box interface (that was developed at the Boise hack). This prepares for that by:

* Removing the old biblebox interface
* Restructuring the web root to simplify the nginx config
* Exposing the contents of the USB drive as UTF-8 encoded json (per the nginx-autoindex module)
* Adding some sample content so we can see UTF-8 filenames, including arabic so we have a RtL example

### CODE REVIEW

File: ansible/roles/webserver-content/tasks/main.yml [hunk 1]
```diff
@@ -34,41 +25,28 @@
...
-
-# This makes the config files executable but
-# it seems like the only other option is to separately
-# set permissions on directory in an additional task
-# Since the config files will probably be going away
-# I think this is fine for now.
-- name: update permissions on biblebox config files
-  file:
-    dest: "{{ biblebox_config_root }}"
+- name: Create placeholder index file
```
Comment (edwinsteele) [inline ansible/roles/webserver-content/tasks/main.yml]: That's correct. You can remove the task that creates the placeholder index file, and replace it with a task like we have for copying/synchronising the admin ui. PM on slack if you're unfamiliar with ansible - happy to talk you through it.

**Comment by edwinsteele** [comment]
Thanks, and yes, the client interface should go under html.

File: ansible/group_vars/all [hunk 2]
```diff
@@ -24,18 +26,25 @@ nginx_vhosts:
...
+    error_page: "404 /index.html"
     access_log: "/var/log/nginx/biblebox-access.log"
     error_log: "/var/log/nginx/biblebox-error.log"
     extra_parameters: |
       rewrite_log on;
 
-      # pass the PHP scripts to FastCGI server listening on 127.0.0.1:9000
+      # Admin interface
+      location /admin/ {
+        root {{ biblebox_web_root }};
```
Comment (furnox) [inline ansible/group_vars/all]: Should this be biblebox_admin_root?

Comment (edwinsteele) [inline ansible/group_vars/all]: Only if we had `alias {{ biblebox_admin_root }};` instead. Given `/admin` is a common suffix, I'm told it's more idiomatic to use a `root` element, even though they're functionally identical.

File: ansible/group_vars/all [hunk 3]
```diff
@@ -46,27 +55,14 @@ nginx_vhosts:
         auth_basic_user_file /usr/local/biblebox/etc/basicauth;
       }
 
-      location /shared/ {
```
Comment (kldavis4) [inline ansible/group_vars/all]: The way I am understanding this, it looks like the fallback to web root content is no longer happening with this change. The intention of that was to allow the device to be shipped with some on device content not on the usb stick that would be merged with whatever is already on the device.

Comment (furnox) [inline ansible/group_vars/all]: I thought the intent of the biblebox itself was that all content could be quickly removed and hidden/destroyed? I'm a little late to the party so I'm sure I missed some discussion.

File: ansible/group_vars/all (hunk 2 above)
Comment (edwinsteele) [inline ansible/group_vars/all]: When you use `root`, the path on disk is the value of the `root` directive and the URL. The url starts with /admin/ in this case, so if we used `{{ biblebox_admin_root }}`, nginx would be looking for files under `/var/www/biblebox/admin/admin`. The `alias` directive doesn't append the URL when nginx is looking for files on disk so `alias {{ biblebox_admin_root }}` is equivalent to what we have. Perhaps you're thinking of `alias`?

File: ansible/group_vars/all (hunk 3 above)
Comment (edwinsteele) [inline ansible/group_vars/all]: The fallback to web root content has been taken away, but I took it out without realising the intent :-\ I think it's nice to have that feature, so let's merge this (so @furnox can continue) and I'll put it back in a followup PR.

File: ansible/group_vars/all (hunk 2 above)
Comment (furnox) [inline ansible/group_vars/all]: No, I'm just not real familiar with nginx configs. :-)
I didn't realize that root appended the url. This makes perfect sense now.

**Comment by edwinsteele** [comment]
I've created https://github.com/edwinsteele/biblebox-pi/issues/29 to track the work to re-instate the ability to display non-USB content alongside USB content.

File: ansible/group_vars/all (hunk 3 above)
Comment (edwinsteele) [inline ansible/group_vars/all]: I'm not sure about the desire to quickly remove/hide content. I haven't heard about that use-case, but it'd be good to ask

**Pull request merged**